apps/omg_watcher/lib/omg_watcher/ethereum_event_aggregator.ex

```diff
@@ -0,0 +1,327 @@
+# Copyright 2019-2020 OmiseGO Pte Ltd
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+defmodule OMG.Watcher.EthereumEventAggregator do
```

Comment:
I have the same comments here than for the Childchain version of this. Now I'm also wondering if it'd make sense to move the log management to `ex_plasma` to de-dup the code a bit.

Reply:
I think the plans for the next engineering cycle is to try and figure out how the split is going to work out - and it's not quite hashed out yet. so I think (and @achiurizo agreed) that this code duplication might just be beneficial for when we're doing the split... by either keeping it this way, putting it into a common umbrella app and merging it together or putting it into a dependency. so let's try to figure out what we should do in the following weeks.